auto_cpufreq/core.py

```diff
@@ -84,6 +84,40 @@ def file_stats():
     auto_cpufreq_stats_file = open(auto_cpufreq_stats_path, "w")
     sys.stdout = auto_cpufreq_stats_file
 
+
+def find_config_file(args_config_file):
+    """
+    Find the config file to use.
+
+    Look for a config file in the following priorization order:
+    1. Command line argument
+    2. User config file
+    3. System config file
+
+    :param args_config_file: Path to the config file provided as a command line argument
+    :return: The path to the config file to use
+    """
+
+    # Prepare paths
+    home = os.getenv("HOME")
+    user_config_dir = os.getenv("XDG_CONFIG_HOME", default=os.path.join(home, ".config"))
+    user_config_file = os.path.join(user_config_dir, "auto-cpufreq/auto-cpufreq.conf")
+    system_config_file = "/etc/auto-cpufreq.conf"
+
+    # If config file is provided as a command line argument but is invalid, exit with error message
+    if args_config_file is not None:
+        if not os.path.isfile(args_config_file):
+            print(f"Config file specified with '--config {args_config_file}' not found.")
+            sys.exit(1)
+
+    if args_config_file and os.path.isfile(args_config_file):  # (1) Command line argument
```

Comment:
`os.path.isfile(args_config_file)` in line 113 is redundant since it would be caught by line 109

Reply:
I've merged the two conditions: https://github.com/AdnanHodzic/auto-cpufreq/pull/672/commits/2b3e9d9574b82c6bfab4f8fb2bcd72646eb5a72b